lib/rest-api.php

```diff
@@ -450,6 +458,48 @@ function gutenberg_handle_early_callback_checks( $response, $handler, $request )
 }
 add_filter( 'rest_request_before_callbacks', 'gutenberg_handle_early_callback_checks', 10, 3 );
 
+/**
+ * Include additional query parameters on the posts query endpoint.
+ *
+ * @see https://core.trac.wordpress.org/ticket/43998
+ *
+ * @param array  $query_params JSON Schema-formatted collection parameters.
+ * @param string $post_type    Post type being accessed.
+ * @return array
+ */
+function gutenberg_filter_post_collection_parameters( $query_params, $post_type ) {
+	$post_types = array( 'page', 'wp_block' );
+	if ( in_array( $post_type->name, $post_types, true )
+		&& isset( $query_params['per_page'] ) ) {
+		// Change from '1' to '-1', which means unlimited.
```

Comment:
I don't quite understand this comment; where is `'1'` being checked for?

Reply:
> where is `'1'` being checked for?

It doesn't need to be checked for. `'1'` is the default value: https://github.com/WordPress/WordPress/blob/2f792d442bf771a3aade170cc9cae459f024c57b/wp-includes/rest-api/endpoints/class-wp-rest-controller.php#L321

Reply:
I don't see any checks at all for `minimum` though.

Maybe I just don't get the query params variable here but it looks like it's a dictionary and here you're just checking for `$query_params['per_page']`, not `$query_params['per_page']['minimum']`.

It would be at least handy to say `// Change from default value '1' to '-1', which means unlimited.`

That said this code strikes me as assuming minimum is set to the default if set at all which I can't tell from here.

Reply:
Oh wait, I understand the code now but it was not obvious at first. I'd suggest changing this function to add a few more comments for context:

```php
function gutenberg_filter_post_collection_parameters( $query_params, $post_type ) {
	$post_types = array( 'page', 'wp_block' );
	// HACK: If 'per_page' is set, ignore it and change the items returned
	// per-page to be '-1'.
	// See: https://github.com/WordPress/gutenberg/issues/6180#issuecomment-384511059
	if ( in_array( $post_type->name, $post_types, true )
		&& isset( $query_params['per_page'] ) ) {
		// Change from '1' (defalut) to '-1', which means unlimited.
		$query_params['per_page']['minimum'] = -1;
		// Default sanitize callback is 'absint', which won't work in our case.
		$query_params['per_page']['sanitize_callback'] = 'rest_sanitize_request_arg';
	}
	return $query_params;
}
```

That seems to be the intent of the code, right?

Reply:
I don't think that comment is right. This is adjusting the collection params schema, not the actual request arguments. The schema ( `per_page`) is validated by WordPress itself in `WP_REST_Request`  using `rest_validate_value_from_schema`.